src/videojs-media-sources.js

```diff
@@ -301,9 +365,9 @@
           offset += segment.byteLength;
         });
 
-        // Set timestampOffset if we have been given one
-        if (this.timestampOffset !== undefined) {
-          destinationBuffer.timestampOffset = this.timestampOffset;
+        // Set timestampOffset if it was changed before this append
```

Comment:
`timestampOffset` is side-effecting and you're shifting that side-effect in time. I don't think I'm going out on a limb to say that lazy IO is confusing. Assuming some amount of laziness is required, my proposal:

- makes the "normal" case strict
- gets rid of an extra piece of state to manage (or screw up)